mscore/prefsdialog.cpp

```diff
@@ -948,7 +949,26 @@ void PreferenceDialog::buttonBoxClicked(QAbstractButton* button)
 
 void PreferenceDialog::apply()
       {
-      advancedWidget->save();
+      // protects against multiple clicks to apply, which make the program look like it has been frozen
+      if (applyTimer.isValid()) {
+            if (applyTimer.elapsed() > applyCoolDown)
+                  applyTimer.restart();
+            else
+                  return;
+            }
+      else {
+            applyTimer.start();
+            }
+
+      if (preferences.globalStyle() != MuseScoreStyleType(styleName->currentIndex()))
+         uiStyleThemeChanged = true;
+
+      std::vector<QString> changedAdvancedProperties = advancedWidget->save();
+
+      for (auto x : changedAdvancedProperties) {
+            if (x.split("/").at(0) == "ui")
```

Comment:
I think it would be better to use the method `QString::startsWith`

Reply:
Nice :-)